_test/test_utilities_herbert/test_stringmatchi.m

```diff
@@ -0,0 +1,124 @@
+classdef test_stringmatchi < TestCase
+    % Test of stringmatchi and stringmatchi_log
+    
+    methods
+        %--------------------------------------------------------------------------
+        function test_nonUnique_unambiguous (~)
+            % Two elements of the cell array start with the test string, but
+            % ambiguity is resolved by one being an exact match
+            strcell = {'hello', 'help', 'hell', 'burp'};
+            ix = stringmatchi ('hell', strcell);
+            ok = stringmatchi_log ('hell', strcell);
+            
+            assertEqual (ix, 3);            
+            assertEqual (ok, logical([0,0,1,0]));            
+        end
+        
+        %--------------------------------------------------------------------------
+        function test_nonUnique_unambiguous_stringInput (~)
+            % Two elements of the cell array start with the test string, but
+            % ambiguity is resolved by one being an exact match
+            strcell = ["hello", "help", "hell", "burp"];
+            ix = stringmatchi ('hell', strcell);
+            ok = stringmatchi_log ('hell', strcell);
+            
+            assertEqual (ix, 3);            
+            assertEqual (ok, logical([0,0,1,0]));            
+        end
+        
+        %--------------------------------------------------------------------------
+        function test_nonUnique_ambiguous (~)
+            % Three elements of the cell array start with the test string, but
+            % no resolution of the ambiguity by an exact match
+            strcell = {'hello', 'help', 'hell', 'burp'};
```

Comment:
Historic naming convention, which is reflected in the use of str_* for these utility routines for many years. Only since R2016b/R2017a has the Matlab string class been around. Previously Mathworks freely used 'string' and 'str' for character vectors (witness the function cellstr as an example). I've noticed they use 'character vector' in the recent documentation now. I'm pretty sure I've seen 'string' when they mean 'character vector' in various places, however.